### PR TITLE
(#604) Update tests to use least privileged users

### LINKED ISSUE #604: Update tests to use least-privledged users
Currently our CMS tests test using the admin user which basically is allowed to do anything. In fact so much so that Drupal usually checked "is the user admin, OR some complicated permissions check". In Drupal a role is a collection of permissions. A user can be granted multiple roles and the permissions in each role map to a specific level of access. For example what we refer to as "making a user a site admin" actually means they have multiple Drupal roles, Author, Editor, Advanced Editor and Site Admin. We did this so that we could not miss things like an editor can edit an article, but we missed the edit article permission for site admin so they cannot. Blame Drupal. 

When a site is created on lower tiers, we 

- Basic author - cannot publish content, but can create content and upload images
   - username: 'author'
   - roles:
      - 'content_author'
      - 'image_manager'
      - 'admin_ui'
- Editor - can publish pages and do everything an author can.
   - username: 'editor'
   - roles:
      - 'content_author'
      - 'content_editor'
      - 'image_manager'
      - 'admin_ui'
- Advanced Editor - Can manage navigation (and blocks?) and do everything an editor can.
  - username: 'adveditor'
  - roles:
     - 'content_author'
     - 'content_editor'
     - 'advanced_editor'
     - 'image_manager'
     - 'admin_ui'
- Site Admin - Can manage users and do some cache clearing, can create new blog series, and anything an Advanced editor can
   - username: 'siteadmin'
   - roles:
      - 'content_author'
      - 'content_editor'
      - 'advanced_editor'
      - 'site_admin'
      - 'image_manager'
      - 'admin_ui'
      - 'blog_manager'
- Preview User - Can just see content, including unpublished content, nothing else
   - username: 'preview'
   - roles:
      - 'content_previewer'

See https://github.com/NCIOCPL/cgov-digital-platform/wiki/Editorial-Workflow for a bit more details, but understanding the page is like 4.5 years old.

### PR DESCRIPTION
**Pull request by KateMashkinaNIH**
Closes #604

### CODE REVIEW

File: cypress/e2e/common/cmsSteps.js [hunk 1]
```diff
@@ -8,8 +8,7 @@ const siteSection = Cypress.env('test_site_section');
 const randomStr = Cypress.env('randomStr');
 const frontEndBaseUrl = Cypress.env('front_end_base_url');
 
-When('user enters credentials', () => {
-    expect(username, 'username was set').to.be.a('string').and.not.be.empty
+When('user enters credentials of {string}', (username) => {
```
Comment (blairlearn) [inline cypress/e2e/common/cmsSteps.js:11]: What's going on here?  This is now using a userid that's passed in from the `.feature` file, but it's followed by a test for whether the **admin** password is present.

File: cypress/e2e/common/cmsSteps.js [hunk 2]
```diff
@@ -547,4 +563,30 @@ And('user selects {string} from Source View dropdown', (sourceDropdown) => {
 
 And('user enters {string} in Raw HTML Content under Secondary Contents', (value) => {
     cy.get('table[id*="field-secondary-contents"]').find('textarea[name*="field_raw_html"]').type(value);
+})
+
+When('user clicks on {string} option button',(deleteBtn)=>{
+cy.get("[id*='edit-delete']").click();
+});
```
Comment (blairlearn) [inline cypress/e2e/common/cmsSteps.js:570]: The `deleteBtn` variable isn't used.  This should just be `'user clicks on {string} option button'`.

File: cypress/e2e/common/cmsSteps.js [hunk 3]
```diff
@@ -547,4 +563,30 @@ And('user selects {string} from Source View dropdown', (sourceDropdown) => {
...
+When('user confirms {string} action',(deleteConfirm)=>{
+    if(deleteConfirm.includes('Delete')){
+        cy.get(`div[class*='ui-dialog-buttonset form-actions'] button:contains('${deleteConfirm}')`).click();
+    }else {
+        cy.get(`input[value='${deleteConfirm}']`).click()
+    }
+
+});
```
Comment (blairlearn) [inline cypress/e2e/common/cmsSteps.js:578]: Is this only deletions?  Or can it be used for other confirmations?  I _think_ this is aimed at addressing English versus Spanish, in which case `deleteConfirm` should be something more descriptive such as `buttonText`.

The hard-coded check for `'Delete'` leaves me wondering whether the condition is needed or if there should just be a single button selection.

**Review by blairlearn — CHANGES_REQUESTED** [review]

File: cypress/e2e/common/cmsSteps.js (hunk 1 above)
Comment (blairlearn) [reply cypress/e2e/common/cmsSteps.js:11]: After discussion with @KateMashkinaNIH, all test accounts use the same password.  This is about the variable name for passing it.

File: cypress/e2e/common/cmsSteps.js (hunk 3 above)
Comment (KateMashkinaNIH) [reply cypress/e2e/common/cmsSteps.js]: addressed ( added a comment to explain why)

File: cypress/e2e/common/cmsSteps.js (hunk 2 above)
Comment (KateMashkinaNIH) [reply cypress/e2e/common/cmsSteps.js]: now it is)

**Comment by KateMashkinaNIH** [comment]
test run https://github.com/NCIOCPL/cgov-digital-platform-acceptance-tests/actions/runs/8163576765/job/22317133341

**Review by blairlearn — APPROVED** [review]

**Pull request merged**